satellite/gc/gc_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gc_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/internal/testrand"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storage/streams"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite"
+)
+
+// TestGarbageCollection does the following:
+// * Set up a network with one storagenode
+// * Upload two objects
+// * Delete one object from the metainfo service on the satellite
+// * Trigger a bloom filter generation
+// * Check that pieces of the deleted object are deleted on the storagenode
+// * Check that pieces of the kept object are not deleted on the storagenode
+func TestGarbageCollection(t *testing.T) {
+	t.Skip("Garbage collection not implemented yet.")
+
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 1, UplinkCount: 1,
+		Reconfigure: testplanet.Reconfigure{
+			Satellite: func(log *zap.Logger, index int, config *satellite.Config) {
+				// TODO see if we should reconfigure anything
+			},
+		},
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+		upl := planet.Uplinks[0]
+		targetNode := planet.StorageNodes[0]
+
+		checker := satellite.Repair.Checker
+		checker.Loop.Stop()
+
+		// Upload two objects
+		testData1 := testrand.Bytes(8 * memory.KiB)
+		testData2 := testrand.Bytes(8 * memory.KiB)
+
+		err := upl.Upload(ctx, satellite, "testbucket", "test/path/1", testData1)
+		require.NoError(t, err)
+		deletedEncPath, pointerToDelete := getPointer(ctx, t, satellite, upl, "testbucket", "test/path/1")
+		deletedPieceID := pointerToDelete.GetRemote().RootPieceId.Derive(targetNode.ID())
+
+		err = upl.Upload(ctx, satellite, "testbucket", "test/path/2", testData2)
+		require.NoError(t, err)
+		_, pointerToKeep := getPointer(ctx, t, satellite, upl, "testbucket", "test/path/2")
+		keptPieceID := pointerToKeep.GetRemote().RootPieceId.Derive(targetNode.ID())
+
+		// Delete object from metainfo service on satellite
+		err = satellite.Metainfo.Service.Delete(ctx, deletedEncPath)
+		require.NoError(t, err)
+
+		// Check that piece of the deleted object is on the storagenode
+		pieceInfo, err := targetNode.DB.PieceInfo().Get(ctx, satellite.ID(), deletedPieceID)
+		require.NoError(t, err)
+		require.NotNil(t, pieceInfo)
+
+		// Trigger bloom filter generation by running checker
+		err = checker.IdentifyInjuredSegments(ctx)
+		require.NoError(t, err)
+
```

Comment:
I am thinking we need to add a sleep/wait timer, why ?because

1. Assuming that the IdentifyInjuredSegments(ctx) handles/implements the bloom filters for each SN. After a bloom filter is created, when does the satellite sends that corresponding bloom filter should be sent to the SNs?
2. After the SN, receives that Bloom Filter thru endpoint, should execute it to remove the pieces accordingly. 

To do all these, my thought is we need to had a wait/sleep timer and then do the PIeceinfo.Get to check if that pieces is removed.